slamd/discovery/processing/algorithms/plot_generator.py

```diff
@@ -67,3 +108,40 @@ def create_tsne_input_space_plot(cls, plot_df):
             )
         )
         return json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
+
+    @classmethod
+    def _create_scatter_plot(cls, x, y, color=None, customdata=None, error_x=None, error_y=None):
+        return go.Scatter(
+            x=x,
+            y=y,
+            mode='markers',
+            # Manually configure marker size and legend and use the default colorscale for px
+            marker=dict(
```

Comment:
Not sure about this one - but maybe there should be small factory functions to create these dicts (and give them names)